carcara/src/checker/parallel/mod.rs

```diff
@@ -434,13 +223,127 @@ impl<'c> ParallelProofChecker<'c> {
+                        .map_err(|e| {
+                            // Signalize to other threads to stop the proof checking
+                            should_abort.store(true, Ordering::Release);
+                            Error::Checker {
+                                inner: e,
+                                rule: step.rule.clone(),
+                                step: step.id.clone(),
+                            }
+                        })?;
+
+                    if step.clause.is_empty() {
+                        self.reached_empty_clause = true;
+                    }
+                }
+                ProofCommand::Subproof(s) => {
+                    let time = Instant::now();
+                    let step_id = command.id();
+
+                    self.context
+                        .push(&mut pool, &s.assignment_args, &s.variable_args)
+                        .map_err(|e| {
+                            // Signalize to other threads to stop the proof checking
+                            should_abort.store(true, Ordering::Release);
+                            Error::Checker {
+                                inner: e.into(),
+                                rule: "anchor".into(),
+                                step: step_id.to_owned(),
+                            }
+                        })?;
+
+                    if let Some(stats) = &mut stats {
+                        // Collects statistics
+                        let rule_name = match s.commands.last() {
+                            Some(ProofCommand::Step(step)) => {
+                                format!("anchor({})", &step.rule)
+                            }
+                            _ => "anchor".to_owned(),
+                        };
+                        stats.results.add_step_measurement(
+                            stats.file_name,
+                            step_id,
+                            &rule_name,
+                            time.elapsed(),
+                        );
+                    }
+                }
+                ProofCommand::Assume { id, term } => {
+                    if !self.check_assume(id, term, &proof.premises, &iter, &mut stats) {
+                        // Signalize to other threads to stop the proof checking
+                        should_abort.store(true, Ordering::Release);
+                        return Err(Error::Checker {
+                            inner: CheckerError::Assume(term.clone()),
+                            rule: "assume".into(),
+                            step: id.clone(),
+                        });
+                    }
+                }
+            }
+            // Verify if any of the other threads found an error and abort in case of positive
+            if should_abort.load(Ordering::Acquire) {
+                break;
+            }
+        }
+
+        // Returns Ok(reached empty clause, isHoley, current thread statistics)
```

Comment:
A small detail: you forgot to update this comment to reflect the new return type
```suggestion
        // Returns Ok(reached empty clause, isHoley)
```